src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,97 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Arrays;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private final int maxItems = 10;
+    private boolean flag = false;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[0];
+        values = (V[]) new Object[0];
+    }
+
+    public StorageImpl(K[] keys, V[] values) {
+        this.keys = keys;
+        this.values = values;
+    }
+
     @Override
     public void put(K key, V value) {
+        K[] tempKeys = (K[]) new Object[keys.length + 1];
+        V[] tempValues = (V[]) new Object[values.length + 1];
+
+        if (keys.length + 1 > maxItems) {
+            System.out.println("You reached items limit");
+            return;
+        }
+
+        for (int i = 0; i < keys.length; i++) {
+            tempKeys[i] = keys[i];
+            tempValues[i] = values[i];
+        }
+
+        for (int i = 0; i < keys.length; i++) {
+            if ((keys[i] == null && key == null) || (keys[i] != null && keys[i].equals(key))) {
```

Comment:
```suggestion
            if (keys[i] == null && key == null || keys[i] != null && keys[i].equals(key)) {
```
redundant brackets. Read about logic operators' priority

Reply:
thank you